inception-external-search-core/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/ExternalSearchHighlight.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2019
+ * Ubiquitous Knowledge Processing (UKP) Lab
+ * Technische Universität Darmstadt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.inception.externalsearch;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExternalSearchHighlight
+{
+
+    private String highlight; // with <em> tags
+
+    private List<KeywordOffset> offsets = new ArrayList<>();
+
+    public ExternalSearchHighlight(String aHighlight) {
+        highlight = aHighlight;
+    }
+
+    public String getHighlight()
+    {
+        return highlight;
+    }
+
+    public void setHighlight(String highlight)
+    {
+        this.highlight = highlight;
+    }
+
+    public List<KeywordOffset> getOffsets() {
+        return offsets;
+    }
+
+    public void addOffset(int start, int end) {
```

Comment:
An `ExternalSearchHighlight` which lacks offsets should be invalid, so I would recommend removing this setter and passing a list of offsets into the constructor (make a defensive copy). This also allows you to check the list is non-empty.  The highlight string could reasonably be absent (`null`; I'd usually make it [`Optional<String>`](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Optional.html) but the project doesn't currently depend on Guava [directly; it is an indirect dependency through four other dependencies: `de.agilecoders.wicket:wicket-bootstrap-core`, `de.tudarmstadt.ukp.inception.rdf4j:rdf4j-sail-lucene-api`, `org.nd4j:nd4j-common`, `org.reflections:reflections`])

Reply:
Changes made accordingly, ready for review.

Reply:
We avoid Guava as a direct dependency. Actually, I wasn't aware that we meanwhile drag it in through so many channels... need to see if that is actually required.

Reply:
The reason I used Guava's Optional was because it is serializable, where as java.util.Optional is not. This class needs to be serializable.

Reply:
Ah, got it. Since we don't want the Guava dependency, just drop use of `Optional` altogether.

Reply:
Removed.